### PR TITLE
Support Exporting GeneralizedRCNNTransform to ONNX

### PR DESCRIPTION
**Pull request by lara-hdr**
Modifications to GeneralizedRCNNTransform to export the image preprocessing (batch_imgs).

The dded test is disabled until Resize is implemented for opset 11 in ONNX Runtime (for now interpolate with bilinear mode is not well supported in PyTorch master, so the test will fail. I tested locally the changes in this PR by changing 'bilinear' mode to 'nearest' (without align_corners) and matching numbers with ORT and analysing the exported model).

### CODE REVIEW

File: torchvision/models/detection/transform.py [hunk 1]
```diff
@@ -88,21 +92,42 @@ def resize(self, image, target):
...
-        max_size[2] = int(math.ceil(float(max_size[2]) / stride) * stride)
-        max_size = tuple(max_size)
-
-        batch_shape = (len(images),) + max_size
-        batched_imgs = images[0].new(*batch_shape).zero_()
-        for img, pad_img in zip(images, batched_imgs):
-            pad_img[: img.shape[0], : img.shape[1], : img.shape[2]].copy_(img)
-
-        return batched_imgs
+        if torch._C._get_tracing_state():
```
Comment (fmassa) [inline torchvision/models/detection/transform.py]: This function has been effectively entirely rewritten to make it work for ONNX. In this case, I think it would make sense to write a new method, `_batch_images_for_onnx`, which is dispatched instead if we are in tracing mode.

thoughts?

File: torchvision/models/detection/transform.py [hunk 2]
```diff
@@ -88,21 +92,42 @@ def resize(self, image, target):
...
+            max_size = tuple(max_size)
+
+        # work around for
+        # pad_img[: img.shape[0], : img.shape[1], : img.shape[2]].copy_(img)
+        # which is not yet supported in onnx
+        if torch._C._get_tracing_state():
+            padded_imgs = ()
+            for img in images:
+                padding = [(s1 - s2) for s1, s2 in zip(max_size, tuple(img.shape))]
+                concat_0 = torch.cat((img, torch.zeros(padding[0], img.shape[1], img.shape[2])), 0)
```
[the remark this answers is not in the document]
Comment (lara-hdr) [reply torchvision/models/detection/transform.py]: That was my first try, the problem is that padding is static in ONNX.

I could move the code related to padding in a separate function, and replace lines 119-122 to something like "padded_img = _onnx_dynamic_pad() " for readability.

File: test/test_onnx.py [hunk 3]
```diff
@@ -35,6 +36,17 @@ def run_model(self, model, inputs):
         # validate the exported model with onnx runtime
         self.ort_validate(onnx_io, inputs, outputs)
 
+        # verify with the other inputs
+        if other_inputs:
```
[the remark this answers is not in the document]
Comment (lara-hdr) [reply test/test_onnx.py]: I thought it would be clearer to separate the input used to export the model.
But a single list would be fine too. 

Will make the change in the next commit.

File: torchvision/models/detection/transform.py (hunk 1 above)
Comment (lara-hdr) [reply torchvision/models/detection/transform.py]: Definitely.
I will include it in the next commit.
